src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -77,22 +77,30 @@ public void Dispose()
         public Task<Result<ComponentResponse[]>> GetComponentsSearchProjectsAsync(ComponentRequest request,
             CancellationToken token)
         {
-            const string SearchProjectsAPI = "api/components/search_projects"; // Since 6.2; internal
-
-            var query = AppendQueryString(SearchProjectsAPI, "?organization={0}&p={1}&ps={2}&asc=true",
-                request.OrganizationKey, request.Page, request.PageSize); // TODO: should handle optional params
+            // Since 6.2; internal
+            var apiPath = BuildRelativeUrl("api/components/search_projects",
+                new Dictionary<string, string>
```

Comment:
I don't like this kind of "serialization", unfortunately there is nothing built-in that will do the job. I think the cleaner approach would be to keep the request objects and use json attributes to specify the query string argument names. Then using json.net fill the values... However, I guess it is good enough for now if it is isolated.

Reply:
Perhaps some day we could write a better version of this http://geeklearning.io/serialize-an-object-to-an-url-encoded-string-in-csharp/